reactor-core/src/main/java/reactor/core/publisher/MonoZip.java

```diff
@@ -333,13 +339,26 @@ public void onNext(Object t) {
 		@Override
 		public void onError(Throwable t) {
 			error = t;
-			parent.signalError(t);
+			if (parent.delayError) {
```

Comment:
if this is done for inlining, remove the parent's `signalError` method which is only ever called from tests. (but I'm not a fan, means that inner has untested side effects on coordinator)